tests/filecheck/transforms/convert_onnx_to_linalg.mlir

```diff
@@ -64,6 +64,15 @@
 // CHECK-NEXT:  %res_gemm_2_1 = linalg.matmul ins(%8, %t12 : tensor<5x10xf32>, tensor<10x3xf32>) outs(%res_gemm_2 : tensor<5x3xf32>) -> tensor<5x3xf32>
 // CHECK-NEXT:  %res_gemm_2_2 = linalg.add ins(%res_gemm_2_1, %10 : tensor<5x3xf32>, tensor<5x3xf32>) outs(%res_gemm_2_1 : tensor<5x3xf32>) -> tensor<5x3xf32>
 
+%t26 = "test.op"(): () ->  (tensor<1x16x14x14xf32>)
+%res_max_pool_single_out = "onnx.MaxPoolSingleOut"(%t26) {onnx_node_name = "/MaxPoolSingleOut", "auto_pad" = "NOTSET", "ceil_mode" = 0 : si64, "kernel_shape" = [3 : i64, 3 : i64],  "dilations" = [1 : i64, 1 : i64],  "pads" = [0 : i64, 0 : i64, 0 : i64, 0 : i64],  "storage_order" = 0 : si64, strides = [3 : i64, 3 : i64]} : (tensor<1x16x14x14xf32>) -> tensor<1x16x4x4xf32>
+
+// CHECK-NEXT:   %t26 = "test.op"() : () -> tensor<1x16x14x14xf32>
+// CHECK-NEXT:  %res_max_pool_single_out = tensor.empty() : tensor<3x3xf32>
+// CHECK-NEXT:  %res_max_pool_single_out_1 = tensor.empty() : tensor<1x16x4x4xf32>
+// CHECK-NEXT:  %res_max_pool_single_out_2 = arith.constant 0.000000e+00 : f64
```

Comment:
this will only store the maximum above 0.0, but we need to include the maximum even if it's negative. This is a little more tricky, we need to initialise the output buffer with -inf. Here's the way that iree-compile does it:
`%cst_0 = arith.constant 0xFF800000 : f32`
This is for a f32, though, I think the f64 version should be `0xFF00000000000000 : f64`

Reply:
http://bartaz.github.io/ieee754-visualization/

Reply:
Top 12 bits to 1